framework/src/mesh/MooseMesh.C

```diff
@@ -1260,16 +1256,27 @@ MooseMesh::detectPairedSidesets()
     minus_z( 0,  0, -1),
     plus_z ( 0,  0,  1);
 
-  // A first-order Lagrange FE for the face.
-  unsigned int dim = getMesh().mesh_dimension();
-  UniquePtr<FEBase> fe_face (FEBase::build(dim,
-                                           FEType(FIRST, LAGRANGE)));
+  // we need to test all element dimensions from dim down to 1
+  const unsigned int dim = getMesh().mesh_dimension();
+
+  // boundary id sets for elements of different dimensions
+  std::vector<std::set<BoundaryID>>
+    minus_x_ids(dim), plus_x_ids(dim),
+    minus_y_ids(dim), plus_y_ids(dim),
+    minus_z_ids(dim), plus_z_ids(dim);
 
-  // Face is assumed to be flat, therefore normal is assumed to be
-  // constant over the face, therefore only compute it at 1 qp.
-  QGauss qface(dim-1, CONSTANT);
-  fe_face->attach_quadrature_rule (&qface);
-  const std::vector<Point> & normals = fe_face->get_normals();
+  std::vector<UniquePtr<FEBase>> fe_faces(dim);
+  std::vector<UniquePtr<QGauss>> qfaces(dim);
+  for (unsigned i = 0; i < dim; ++i)
+  {
+    // Face is assumed to be flat, therefore normal is assumed to be
+    // constant over the face, therefore only compute it at 1 qp.
+    qfaces[i] = UniquePtr<QGauss>(new QGauss(i, CONSTANT));
+
+    // A first-order Lagrange FE for the face.
+    fe_faces[i] = UniquePtr<FEBase>(FEBase::build(i + 1, FEType(FIRST, LAGRANGE)));
```

Comment:
What's up with the creation of a `UniquePtr` here?  Isn't that already the return value of `FEBase::build()`?